dbt_semantic_interfaces/dbt_semantic_interfaces/protocols/metadata.py

```diff
@@ -0,0 +1,19 @@
+from __future__ import annotations
+
+from typing import Protocol
+
+
+class FileSlice(Protocol):
+    """Provides file slice level context about what something was created from"""
+
+    filename: str
+    content: str
+    start_line_number: int
+    end_line_number: int
+
+
+class Metadata(Protocol):
+    """Provides file context about what something was created from"""
+
+    repo_file_path: str
+    file_slice: FileSlice
```

Comment:
I suspect these should be optional. I think it might be fine to start with them as required and change it later but I honestly don't know.

Were you thinking about making this a contained struct? I seem to remember you saying something about this.